app/src/main/java/org/cru/godtools/ui/dashboard/tools/ToolsCategoryDataModel.kt

```diff
@@ -0,0 +1,29 @@
+package org.cru.godtools.ui.dashboard.tools
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import org.ccci.gto.android.common.androidx.lifecycle.combineWith
+import org.ccci.gto.android.common.db.Expression.Companion.constants
+import org.ccci.gto.android.common.db.Query
+import org.ccci.gto.android.common.db.getAsLiveData
+import org.cru.godtools.model.Tool
+import org.keynote.godtools.android.db.Contract.ToolTable
+import org.keynote.godtools.android.db.GodToolsDao
+
+@HiltViewModel
+class ToolsCategoryDataModel @Inject constructor(dao: GodToolsDao) : ViewModel() {
+    private val tools = Query.select<Tool>().where(
+        ToolTable.FIELD_TYPE.`in`(*constants(Tool.Type.TRACT, Tool.Type.ARTICLE, Tool.Type.CYOA))
+            .and(ToolTable.FIELD_HIDDEN.ne(true))
+    ).orderBy(ToolTable.COLUMN_DEFAULT_ORDER).getAsLiveData(dao)
+
+    val categories = tools.map { it.mapNotNull { it.category }.distinct() }
+    val selectedCategory = MutableLiveData<String?>(null)
```

Comment:
@gyasistory have you done anything with `SavedStateHandle` in a ViewModel yet? this is a simple place to utilize it to persist the currently selected category through app relaunches.

I can point you to a couple examples in our codebase if needed

Reply:
@frett  no I have done anything with that before.

Reply:
k, let's jump on a quick call